hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -171,24 +171,32 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, err
 	}
 
-	result, err := r.reconcile(ctx, hcluster, nodePool)
-	if err != nil {
-		log.Error(err, "Failed to reconcile NodePool")
-		r.recorder.Eventf(nodePool, corev1.EventTypeWarning, "ReconcileError", "%v", err)
-		if err := patchHelper.Patch(ctx, nodePool); err != nil {
-			log.Error(err, "failed to patch")
-			return ctrl.Result{}, fmt.Errorf("failed to patch: %w", err)
+	result := ctrl.Result{}
+	if isPaused, duration := supportutil.IsReconciliationPaused(log, nodePool.Spec.PausedUntil); isPaused {
+		log.Info("Reconciliation paused", "name", req.NamespacedName, "pausedUntil", *nodePool.Spec.PausedUntil)
+		result = ctrl.Result{RequeueAfter: duration}
+	} else {
+		result, err = r.reconcile(ctx, hcluster, nodePool)
+		if err != nil {
```

Comment:
Note that this would prevent the underlying primitives from keep reconciling, e.g capi MachineDeployment which we might want to consider pausing as well (they have the API means for it)

Reply:
Ah I missed that :man_facepalming:  

Yeah we need to ensure the CAPI resources don't change, so we can safely do the backup/restore for migration - I'll go figure out how to pause the underlying resources, thanks!

Reply:
The latest version pauses the MachineDeployment and MachineSet - I used the annotation based on the information you added in https://github.com/kubernetes-sigs/cluster-api/issues/6966 (and added a FIXME so we can potentially switch to the spec field in future)